### PR TITLE
CLDR-13829 Removing old LDML tools and other unused classes.

### PR DESCRIPTION
**Pull request by hagbard**
This is optional, but I think we want to remove the confusion of having two tools which produce different output. There's more unused code in the icu package which can be added here or done as a separate PR.

##### Checklist

- [x] Issue filed: https://unicode-org.atlassian.net/browse/CLDR-13829
- [x] Updated PR title and link in previous line to include Issue number

### CODE REVIEW

**Comment by hagbard** [comment]
Hmm, I included those because I thought Steven Lumis had said they were obsolete.Since it's not clear, I'll split those out into a different PR.

**Comment by hagbard** [comment]
If you remove the LDML converter, you break (at compilation time):
* GenerateCldrCollationTests.java
Since that uses CollationMapper (old LDML tools). If you remove that, you break:
* GenerateCldrTests.java
Which in turn breaks:
* GenerateCldrDateTimeTests.java
* GenerateStatistics.java

The GenerateStatistics dependency could be broken and we could keep that (it's just calling one static method on GenerateCldrTests that probably should have never been there in the first place).
However the 3 other classes would need severe refactoring to be kept in any meaningful way (they'd need to be moved to ICU for a start).

@srl295 can comment more, but I think if they are needed, they're already "broken" since they rely on unmaintained code.

**Comment by hagbard** [comment]
Ohh god, git has wrecked this PR, I'll probably have to kill it and start again :(

**Comment by hagbard** [comment]
Hmm, I think I recovered it. PTAL.

**Comment by hagbard** [comment]
As agreed in the meeting, you (Mark) were going to raise an issue to track this before I delete the generate tests classes (the statistics class is no longer being deleted, just fixed to no longer rely on the others).

**Comment by hagbard** [comment]
TL;DR: This is ready for approval/merging unless anyone has reasons to believe we shouldn't be doing this.

I checked "unicodetools" and nothing in there imports anything from "org.unicode.cldr.icu" or "org.unicode.cldr.ant".

There are 2 uses of the "icu" package name in JSP files however, but they just wildcard the contents without referring to any specific class (and one is just a comment). Thus I think it's completely safe to remove the ICU classes in this PR.

----
[~/i18n/unicodetools]> fgrep -R ".cldr.icu."
UnicodeJsps/WebContent/header.jsp:
unicodetools/org/unicode/jsp/header.jsp:<%@ page import="org.unicode.cldr.icu.*" %>
---
[~/i18n/unicodetools]> fgrep -R ".cldr.ant."
<no output>
----

**Comment by hagbard** [comment]
Since it's been a week and nobody else has mentioned any objections, I'll merge this.
